src/key/factory.js

```diff
@@ -161,23 +152,23 @@ async function wrapKeyObject(secretKeyPacket, secretSubkeyPackets, options, conf
       enums.symmetric.aes256,
       enums.symmetric.aes128,
       enums.symmetric.aes192
-    ], config.encryptionCipher);
+    ], config.preferredCipherAlgorithm);
     if (config.aeadProtect) {
       signaturePacket.preferredAeadAlgorithms = createdPreferredAlgos([
         enums.aead.eax,
         enums.aead.ocb
-      ], config.aeadMode);
+      ], config.preferredAeadAlgorithm);
     }
     signaturePacket.preferredHashAlgorithms = createdPreferredAlgos([
       // prefer fast asm.js implementations (SHA-256)
       enums.hash.sha256,
       enums.hash.sha512
-    ], config.preferHashAlgorithm);
+    ], config.preferredHashAlgorithm);
     signaturePacket.preferredCompressionAlgorithms = createdPreferredAlgos([
+      enums.compression.uncompressed,
       enums.compression.zlib,
-      enums.compression.zip,
-      enums.compression.uncompressed
-    ], config.compression);
+      enums.compression.zip
```

Comment:
This is very nitpicky, but if someone sets `preferredCompressionAlgorithm = zlib`, this would result in `zlib, uncompressed, zip`, which is slightly strange, so I would actually leave the order here as-is, and leave it up to `createPreferredAlgos` to hoist `uncompressed` to the front, if needed.